src/main.js

```diff
@@ -104,120 +123,138 @@ export default class Auth {
 	}
 
 	/**
-	 * Make post request to a specific endpoint, and return the response.
-	 * @param {string} endpoint The name of the endpoint.
+	 * Makes post request to a specific endpoint, and return the response.
+	 * @param {string} endpoint Name of the endpoint.
 	 * @param {any} request Body to pass to the request.
 	 * @private
 	 */
-	api(endpoint, body) {
+	async api(endpoint, body) {
```

Comment:
No need to convert into an async function if we return a promise.
I'm not sure the engine optimizes the unnecessary promise out or not.

Reply:
Should not the method be async because it has await calls inside?